notebook2/notebook/notebook.py

```diff
@@ -200,62 +335,62 @@ def root():
+                               form_action_url=external_url_for('notebook'),
                                images=list(WORKER_IMAGES),
                                default='hail')
-    svc_name = session['svc_name']
-    jupyter_token = session['jupyter_token']
-    log.info('redirecting to ' + external_url_for(f'instance/{svc_name}/?token={jupyter_token}'))
-    return redirect(external_url_for(f'instance/{svc_name}/?token={jupyter_token}'))
 
+    # TODO: We currently simplify state tracking, since using vanilla js makes this trickier
+    # by restricting to one alive notebook displayed
+    # This could be problematic if multiple notebooks are found in running state
+    session['notebook'] = notebooks[0]
+
+    return render_template('notebook.html', notebook=notebooks[0])
 
-@app.route('/new', methods=['GET'])
+
+@app.route('/notebook/delete', methods=['POST'])
 @requires_auth()
-def new_get():
-    pod_name = session.get('pod_name')
-    svc_name = session.get('svc_name')
-    if pod_name:
-        delete_worker_pod(pod_name, svc_name)
-        del session['pod_name']
+def notebook_delete():
+    notebook = session.get('notebook')
 
-    if svc_name is not None:
-        del session['svc_name']
+    if notebook is not None:
+        delete_worker_pod(notebook['pod_name'], notebook['svc_name'])
+        del session['notebook']
 
     return redirect(external_url_for('notebook'))
 
 
-@app.route('/new', methods=['POST'])
+@app.route('/notebook', methods=['POST'])
 @requires_auth()
-def new_post():
-    log.info('new received')
+def notebook_post():
     image = request.form['image']
+
     if image not in WORKER_IMAGES:
-        return '403 Forbidden', 403
-    jupyter_token = uuid.uuid4().hex  # FIXME: probably should be cryptographically secure
-    svc, pod = start_pod(jupyter_token, WORKER_IMAGES[image])
-    session['svc_name'] = svc.metadata.name
-    session['pod_name'] = pod.metadata.name
-    session['jupyter_token'] = jupyter_token
-    return redirect(external_url_for(f'wait'))
+        return '', 404
 
+    jupyter_token = uuid.uuid4().hex
+    name = request.form.get('name', 'a_notebook')
+    safe_user_id = user_id_transform(session['user']['id'])
 
-@app.route('/wait', methods=['GET'])
-@requires_auth()
-def wait_webpage():
-    return render_template('wait.html')
+    pod = start_pod(jupyter_token, WORKER_IMAGES[image], name, safe_user_id)
+    session['notebook'] = marshall_notebooks([pod], svc_status = "Running")[0]
+
+    return redirect(external_url_for('notebook'))
 
 
 @app.route('/auth/<requested_svc_name>')
 @requires_auth()
 def auth(requested_svc_name):
-    approved_svc_name = session.get('svc_name')
-    if approved_svc_name and approved_svc_name == requested_svc_name:
+    notebook = session.get('notebook')
+
+    if notebook is not None and notebook['svc_name'] == requested_svc_name:
         return '', 200
-    return '', 403
+
+    return '', 404
```

Comment:
since we always return 403 or 200 from this endpoint, there's no leaked bit of information and 403 seems more semantically correct for something called `auth` (though it ought to be called authorized in retrospect).